// igniteui-webcomponents/components/button/themes/button/light/button.indigo.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: max(var(--is-large, 1) * 2.25rem, var(--is-medium, 1) * 2rem, var(--is-small, 1) * 1.75rem);
}

[part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(var(--size) / 2)), calc(var(--size) / 2));
}
[part=base]::before, [part=base]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * calc(var(--size) / 2)), calc(var(--size) / 2));
}

:host([variant=fab]) {
  --size: max(var(--is-large, 1) * 3rem, var(--is-medium, 1) * 2.625rem, var(--is-small, 1) * 2.25rem);
}

:host(:not([disabled])[variant=flat]) [part=base] {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  -webkit-transition: color 0.15s ease-in-out, -webkit-box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out, -webkit-box-shadow 0.15s ease-in-out;
  -o-transition: color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out, box-shadow 0.15s ease-in-out, -webkit-box-shadow 0.15s ease-in-out;
}
:host(:not([disabled])[variant=flat]) [part=base]:hover, :host(:not([disabled])[variant=flat]) [part=base]:focus, :host(:not([disabled])[variant=flat]) [part=base]:active {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}
:host(:not([disabled])[variant=flat]) [part=base]:hover::before {
  background: transparent;
}
:host(:not([disabled])[variant=flat]) [part=base]:focus::before, :host(:not([disabled])[variant=flat]) [part=base]:active::before {
  background: transparent;
}
:host(:not([disabled])[variant=flat]) [part=base]:focus, :host(:not([disabled])[variant=flat]) [part=base]:active {
  -webkit-box-shadow: 0 0 0 0.1875rem hsla(var(--ig-gray-300), var(--ig-gray-a));
          box-shadow: 0 0 0 0.1875rem hsla(var(--ig-gray-300), var(--ig-gray-a));
}

:host(:not([disabled])[variant=outlined]) [part=base] {
  color: hsla(var(--ig-gray-700), var(--ig-gray-a));
  -webkit-transition: color 0.15s ease-out;
  -o-transition: color 0.15s ease-out;
  transition: color 0.15s ease-out;
  -webkit-box-shadow: inset 0 0 0 2px hsla(var(--ig-gray-500), var(--ig-gray-a));
          box-shadow: inset 0 0 0 2px hsla(var(--ig-gray-500), var(--ig-gray-a));
  overflow: visible;
}
:host(:not([disabled])[variant=outlined]) [part=base]::after {
  position: absolute;
  content: "";
  width: 100%;
  height: 100%;
  -webkit-transition: -webkit-box-shadow 0.15s ease-out;
  transition: -webkit-box-shadow 0.15s ease-out;
  -o-transition: box-shadow 0.15s ease-out;
  transition: box-shadow 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  background: hsla(var(--ig-gray-50), var(--ig-gray-a));
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus, :host(:not([disabled])[variant=outlined]) [part=base]:active {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus::before, :host(:not([disabled])[variant=outlined]) [part=base]:active::before {
  background: transparent;
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus::after, :host(:not([disabled])[variant=outlined]) [part=base]:active::after {
  -webkit-box-shadow: 0 0 0 0.1875rem hsla(var(--ig-gray-300), var(--ig-gray-a));
          box-shadow: 0 0 0 0.1875rem hsla(var(--ig-gray-300), var(--ig-gray-a));
}

:host([disabled][variant=fab]) [part=base],
:host(:not([disabled])[variant=fab]) [part=base],
:host([disabled][variant=contained]) [part=base],
:host(:not([disabled])[variant=contained]) [part=base] {
  -webkit-box-shadow: var(--ig-elevation-0);
          box-shadow: var(--ig-elevation-0);
}
:host([disabled][variant=fab]) [part=base]:hover,
:host(:not([disabled])[variant=fab]) [part=base]:hover,
:host([disabled][variant=contained]) [part=base]:hover,
:host(:not([disabled])[variant=contained]) [part=base]:hover {
  -webkit-box-shadow: var(--ig-elevation-0);
          box-shadow: var(--ig-elevation-0);
}

:host(:not([disabled])[variant=fab]) [part=base],
:host(:not([disabled])[variant=contained]) [part=base] {
  color: var(--ig-primary-600-contrast);
}
:host(:not([disabled])[variant=fab]) [part=base]:hover,
:host(:not([disabled])[variant=contained]) [part=base]:hover {
  color: var(--ig-primary-600-contrast);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus, :host(:not([disabled])[variant=fab]) [part=base]:active,
:host(:not([disabled])[variant=contained]) [part=base]:focus,
:host(:not([disabled])[variant=contained]) [part=base]:active {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: 0 0 0 0.1875rem hsla(var(--ig-primary-200), var(--ig-primary-a));
          box-shadow: 0 0 0 0.1875rem hsla(var(--ig-primary-200), var(--ig-primary-a));
}
:host(:not([disabled])[variant=fab]) [part=base]::before,
:host(:not([disabled])[variant=contained]) [part=base]::before {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  opacity: 1;
}
:host(:not([disabled])[variant=fab]) [part=base]:hover::before, :host(:not([disabled])[variant=fab]) [part=base]:focus::before, :host(:not([disabled])[variant=fab]) [part=base]:active::before,
:host(:not([disabled])[variant=contained]) [part=base]:hover::before,
:host(:not([disabled])[variant=contained]) [part=base]:focus::before,
:host(:not([disabled])[variant=contained]) [part=base]:active::before {
  background: hsla(var(--ig-primary-400), var(--ig-primary-a));
  opacity: 1;
}

:host([disabled][variant=flat]) [part=base],
:host([disabled][variant=outlined]) [part=base] {
  background: transparent;
  -webkit-box-shadow: none;
          box-shadow: none;
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}

:host([disabled][variant=fab]) [part=base],
:host([disabled][variant=contained]) [part=base] {
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}`;
