// igniteui-webcomponents/components/button/themes/button/light/button.bootstrap.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: max(var(--is-large, 1) * 3rem, var(--is-medium, 1) * 2.375rem, var(--is-small, 1) * 1.9375rem);
}

:host [part=base] {
  -webkit-transition: all 0.15s ease-out;
  -o-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
}
:host [part=base]::before {
  -webkit-transition: all 0.15s ease-out;
  -o-transition: all 0.15s ease-out;
  transition: all 0.15s ease-out;
}

:host([disabled][variant=flat]) [part=base] {
  background: transparent;
  color: hsla(var(--ig-primary-100), var(--ig-primary-a));
}

:host([disabled][variant=fab]) [part=base],
:host([disabled][variant=contained]) [part=base] {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}

:host([disabled][variant=outlined]) [part=base] {
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-box-shadow: 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
          box-shadow: 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
  opacity: 0.65;
  background: transparent;
}

:host(:not([disabled])[variant=flat]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host(:not([disabled])[variant=flat]) [part=base]:hover {
  color: hsla(var(--ig-primary-800), var(--ig-primary-a));
}
:host(:not([disabled])[variant=flat]) [part=base]:focus, :host(:not([disabled])[variant=flat]) [part=base]:active {
  color: hsla(var(--ig-primary-600), var(--ig-primary-a));
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
}
:host(:not([disabled])[variant=flat]) [part=base]:hover::before {
  background: transparent;
}
:host(:not([disabled])[variant=flat]) [part=base]:focus::before, :host(:not([disabled])[variant=flat]) [part=base]:active::before {
  background: transparent;
  opacity: 1;
}

:host(:not([disabled])[variant=contained]) [part=base] {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: var(--ig-elevation-0);
          box-shadow: var(--ig-elevation-0);
}
:host(:not([disabled])[variant=contained]) [part=base]:hover {
  color: var(--ig-primary-600-contrast);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus, :host(:not([disabled])[variant=contained]) [part=base]:active {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
}
:host(:not([disabled])[variant=contained]) [part=base]::before {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  opacity: 1;
}
:host(:not([disabled])[variant=contained]) [part=base]:hover::before, :host(:not([disabled])[variant=contained]) [part=base]:focus::before, :host(:not([disabled])[variant=contained]) [part=base]:active::before {
  background: hsla(var(--ig-primary-600), var(--ig-primary-a));
  opacity: 1;
}

:host(:not([disabled])[variant=outlined]) [part=base] {
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
  -webkit-box-shadow: 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
          box-shadow: 0 0 0 1px hsla(var(--ig-primary-500), var(--ig-primary-a));
  overflow: visible;
}
:host(:not([disabled])[variant=outlined]) [part=base]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.25rem), 0.25rem);
  position: absolute;
  content: "";
  width: 100%;
  height: 100%;
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover {
  background: hsla(var(--ig-primary-600), var(--ig-primary-a));
  color: var(--ig-primary-600-contrast);
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover::before {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  opacity: 1;
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus::before, :host(:not([disabled])[variant=outlined]) [part=base]:active::before {
  background: transparent;
  opacity: 1;
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus::after, :host(:not([disabled])[variant=outlined]) [part=base]:active::after {
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
}

:host([variant=fab][size=large]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 1rem), 1rem);
}

:host([variant=fab][size=medium]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.75rem), 0.75rem);
}

:host([variant=fab][size=small]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.5rem), 0.5rem);
}

:host(:not([disabled])[variant=fab]) [part=base] {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: var(--ig-elevation-0);
          box-shadow: var(--ig-elevation-0);
}
:host(:not([disabled])[variant=fab]) [part=base]:hover {
  color: var(--ig-primary-600-contrast);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus, :host(:not([disabled])[variant=fab]) [part=base]:active {
  color: var(--ig-primary-600-contrast);
  -webkit-box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
          box-shadow: 0 0 0 0.25rem hsla(var(--ig-primary-200), 0.5);
}
:host(:not([disabled])[variant=fab]) [part=base]::before {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  opacity: 1;
}
:host(:not([disabled])[variant=fab]) [part=base]:hover::before, :host(:not([disabled])[variant=fab]) [part=base]:focus::before, :host(:not([disabled])[variant=fab]) [part=base]:active::before {
  background: hsla(var(--ig-primary-600), var(--ig-primary-a));
  opacity: 1;
}`;
