// igniteui-webcomponents/components/button/themes/button/light/button.fluent.css.js
import { css } from 'lit';
export const styles = css `/* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */ /* stylelint-disable max-line-length */
:host {
  --size: max(var(--is-large, 1) * 2.375rem, var(--is-medium, 1) * 2rem, var(--is-small, 1) * 1.5rem);
}

[part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.125rem), 0.125rem);
  padding: 0 max(var(--is-large, 1) * 16px * var(--ig-spacing-large, var(--ig-spacing, --ig-spacing)), var(--is-medium, 1) * 16px * var(--ig-spacing-medium, var(--ig-spacing, --ig-spacing)), var(--is-small, 1) * 8px * var(--ig-spacing-small, var(--ig-spacing, --ig-spacing)));
}

:host(:not([disabled])[variant=flat]) [part=base] {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}
:host(:not([disabled])[variant=flat]) [part=base]:hover {
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host(:not([disabled])[variant=flat]) [part=base]:active {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}
:host(:not([disabled])[variant=flat]) [part=base]:hover::before, :host(:not([disabled])[variant=flat]) [part=base]:focus::before, :host(:not([disabled])[variant=flat]) [part=base]:active::before {
  background: transparent;
}

:host(:not([disabled])[variant=outlined]) [part=base] {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
  -webkit-transition: color 0.15s ease-out;
  -o-transition: color 0.15s ease-out;
  transition: color 0.15s ease-out;
  -webkit-box-shadow: 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
          box-shadow: 0 0 0 1px hsla(var(--ig-gray-700), var(--ig-gray-a));
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover::before, :host(:not([disabled])[variant=outlined]) [part=base]:focus:hover::before {
  opacity: 1;
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus::before {
  opacity: 1;
  background: transparent;
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus:active::before {
  opacity: 1;
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover, :host(:not([disabled])[variant=outlined]) [part=base]:focus, :host(:not([disabled])[variant=outlined]) [part=base]:active {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}
:host(:not([disabled])[variant=outlined]) [part=base]:active::before, :host(:not([disabled])[variant=outlined]) [part=base]:focus:active::before, :host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:hover:active::before, :host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:active::before {
  opacity: 1;
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:hover::before {
  opacity: 1;
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
}

:host(:not([disabled])[variant=flat]) [part=base]:focus-visible,
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible {
  position: relative;
}
:host(:not([disabled])[variant=flat]) [part=base]:focus-visible::after,
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible::after {
  content: "";
  -webkit-box-shadow: 0 0 0 1px hsla(var(--ig-gray-800), var(--ig-gray-a));
          box-shadow: 0 0 0 1px hsla(var(--ig-gray-800), var(--ig-gray-a));
  position: absolute;
  top: 2px;
  left: 2px;
  pointer-events: none;
  width: calc(100% - 4px);
  height: calc(100% - 4px);
}
:host(:not([disabled])[variant=flat]) [part=base]:focus-visible::before,
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible::before {
  background: transparent;
}

:host(:not([disabled])[variant=contained]) [part=base],
:host(:not([disabled])[variant=fab]) [part=base] {
  color: var(--text-color, var(--ig-primary-600-contrast));
  -webkit-box-shadow: var(--ig-elevation-0);
          box-shadow: var(--ig-elevation-0);
}
:host(:not([disabled])[variant=contained]) [part=base]:hover, :host(:not([disabled])[variant=contained]) [part=base]:focus-visible:hover,
:host(:not([disabled])[variant=fab]) [part=base]:hover,
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible:hover {
  color: var(--text-color, var(--ig-primary-600-contrast));
}
:host(:not([disabled])[variant=contained]) [part=base]:hover::before, :host(:not([disabled])[variant=contained]) [part=base]:focus-visible:hover::before,
:host(:not([disabled])[variant=fab]) [part=base]:hover::before,
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible:hover::before {
  opacity: 1;
  background: var(--hover-background, hsla(var(--ig-primary-600), var(--ig-primary-a)));
}
:host(:not([disabled])[variant=contained]) [part=base]:focus, :host(:not([disabled])[variant=contained]) [part=base]:active,
:host(:not([disabled])[variant=fab]) [part=base]:focus,
:host(:not([disabled])[variant=fab]) [part=base]:active {
  color: var(--text-color, var(--ig-primary-600-contrast));
}
:host(:not([disabled])[variant=contained]) [part=base]::before,
:host(:not([disabled])[variant=fab]) [part=base]::before {
  opacity: 1;
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
}
:host(:not([disabled])[variant=contained]) [part=base]:active::before, :host(:not([disabled])[variant=contained]) [part=base]:focus-visible:active::before,
:host(:not([disabled])[variant=fab]) [part=base]:active::before,
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible:active::before {
  opacity: 1;
  background: var(--active-background, hsla(var(--ig-primary-700), var(--ig-primary-a)));
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible,
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible {
  position: relative;
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible::after,
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible::after {
  content: "";
  -webkit-box-shadow: 0 0 0 1px var(--ig-primary-600-contrast);
          box-shadow: 0 0 0 1px var(--ig-primary-600-contrast);
  position: absolute;
  top: 3px;
  left: 3px;
  pointer-events: none;
  width: calc(100% - 6px);
  height: calc(100% - 6px);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible::before,
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible::before {
  background: transparent;
}

:host([variant=fab]) [part=base] {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.875rem), 0.875rem);
}
:host([variant=fab]) [part=base]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor) * 0.75rem), 0.75rem);
}

:host([disabled][variant=flat]) [part=base] {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
  background: transparent;
}

:host([disabled][variant=fab]) [part=base],
:host([disabled][variant=contained]) [part=base] {
  color: hsla(var(--ig-gray-400), var(--ig-gray-a));
}
:host([disabled][variant=fab]) [part=base]::before,
:host([disabled][variant=contained]) [part=base]::before {
  opacity: 1;
  background: hsla(var(--ig-gray-100), var(--ig-gray-a));
}

:host([disabled][variant=outlined]) [part=base] {
  -webkit-box-shadow: none;
          box-shadow: none;
  color: hsla(var(--ig-gray-500), 8);
  background: hsla(var(--ig-gray-200), var(--ig-gray-a));
}`;
